File: Android/app/src/main/java/com/khsm/app/data/entities/News.java
package com.khsm.app.data.entities;

import java.util.Date;

public class News {
    public final Integer id;
    public final User user;
    public final String text;
    public final Date dateAndTime;

    public News(Integer id, User user, String text, Date dateAndTime) {
        this.id = id;
        this.user = user;
        this.text = text;
        this.dateAndTime = dateAndTime;
    }
}

File: Backend/Data/Entities/News.cs
using System;
using System.ComponentModel.DataAnnotations;

namespace Backend.Data.Entities
{
    public class News
    {
        [Required]
        public int Id { get; set; }
        [Required]
        public User User { get; set; }
        [Required]
        public String Text { get; set; }
        [Required]
        public DateTime? DateAndTime { get; set; }
        
        protected bool Equals(News other)
        {
            return Id == other.Id;
        }
        
        public override bool Equals(object obj)
        {
            if (ReferenceEquals(null, obj)) return false;
            if (ReferenceEquals(this, obj)) return true;
            if (obj.GetType() != this.GetType()) return false;
            return Equals((News) obj);
        }
        
        public override int GetHashCode()
        {
            return Id;
        }
    
    }
    
}
 
 
    